src/plugins/discover/public/application/angular/context/api/utils/get_es_query_sort.ts

```diff
@@ -14,11 +14,18 @@ import { EsQuerySortValue, SortDirection } from '../../../../../kibana_services'
  * @param timeField
  * @param tieBreakerField
  * @param sortDir
+ * @param nanos
  */
 export function getEsQuerySort(
   timeField: string,
   tieBreakerField: string,
-  sortDir: SortDirection
+  sortDir: SortDirection,
+  nanos?: string
 ): [EsQuerySortValue, EsQuerySortValue] {
-  return [{ [timeField]: sortDir }, { [tieBreakerField]: sortDir }];
+  const time = {} as Record<string, any>;
+  time[timeField] = {
+    order: sortDir,
+    format: nanos ? 'strict_date_optional_time_nanos' : 'strict_date_optional_time',
+  };
+  return [{ ...time }, { [tieBreakerField]: sortDir }];
```

Comment:
To omit `any` we could introduce a new `searchSource` interface like

```
export interface SortDirectionFormat {
  order: SortDirection;
  format?: string;
}
export type EsQuerySortValue = Record<string, SortDirection | SortDirectionNumeric | SortDirectionFormat >;
```

or a local one like:

```
export type EsQuerySortValueContext = Record<
  string,
  {
    order: SortDirection;
    format: string;
  }
>;
```

Then the return could look like this
```
return [
    {
      [timeField]: {
        order: sortDir,
        format: nanos ? 'strict_date_optional_time_nanos' : 'strict_date_optional_time',
      },
    },
    { [tieBreakerField]: sortDir },
  ];
```